### PR TITLE
BREAKING CHANGE: Update and streamline SDKs

### PR DESCRIPTION
**Pull request by s-ashwinkumar**
Asana: https://app.asana.com/0/1200041805723615/1200544489692892/f

### CODE REVIEW

File: lib/smartcar-error.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      this.resolution = body.resolution;
+    }
+
+    // Now dynamically set the remaining ones if passed
+    fields.forEach((item) => {
+      if (body[item]) {
+        this[item] = body[item];
+      }
+    });
+    /**
```
Comment (s-ashwinkumar) [inline lib/smartcar-error.js:45]: Not sure about these docs yet. Will check...

File: lib/smartcar-service.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+          response: {
+            body: response.body,
+            headers: {
+              ...headers,
+              ...response.headers,
+            },
+          },
+        });
+      };
+      // add build error part here once errors are done
```
Comment (s-ashwinkumar) [inline lib/smartcar-service.js:70]: remove this line.
```suggestion
```

File: index.js [hunk 3]
```diff
@@ -81,53 +95,116 @@ smartcar.isExpired = function(expiration) {
...
+/**
+ * Verify webhook payload with AMT and signature.
+ *
+ * @method
+ * @param {String} amt - Application Management Token
+ * @param {String} signature - sc-signature header value
+ * @param {object} body - webhook response body
+ * @return {Boolean} true if signature matches the hex digest of amt and body
+ */
+smartcar.verfifyPayload = (amt, signature, body) => (
```
Comment (nbry) [inline index.js:206]: Typo for `verifyPayload` static method

File: test/end-to-end/vehicle.js [hunk 4]
```diff
@@ -218,6 +218,36 @@ test('vehicle permissions', async(t) => {
   t.is(response.meta.requestId.length, 36);
 });
 
+test('vehicle subscribe and unsubscribe - success', async(t) => {
+  const webhookId = 'fdc63623-cd65-4a7f-8ef4-44985f9d7355';
```
Comment (nbry) [inline test/end-to-end/vehicle.js:222]: Gurpreet made environment variables in Travis for testing webhooks! 

`E2E_SMARTCAR_AMT` and `E2E_SMARTCAR_WEBHOOK_ID`

**Comment by nbry** [comment]
I did a review of the SDK, and it looks pretty good to me! Your method map is really cool. 

Let's wait for @rsimari 's review and see what he thinks.

**Comment by nbry** [comment]
Committed a change to smartcar-service.js, that allows for a better naming when a user sends a batch request that includes a request to vehicle attributes. In other words, the batch response for attributes will be `batch.attributes()` instead of `batch['']()`

**Comment by nbry** [comment]
Review #2: @s-ashwinkumar , I did another run-through of the node-sdk. There was a change on Friday to the Smartcar Connect flow, where upon choosing permissions for your car, a "success" page is displayed. However, now you have to press 'continue' to get the url with the access code in the params. For clarity, the continue button has an id of `continue-button`

**Review by nbry — APPROVED** [review]
Other than setting up the tests so they run with the new Smartcar Connect flow, I think this good to go!

**Pull request merged**